### PR TITLE
Strategy should use better param description

### PR DESCRIPTION
**Pull request by ivarconr**
Adds support for more fields sucha as description, required, etc.

Required tasks:
- [x] Implement db-migrations
- [x] Validate input when defining a new strategy
- [x] Write a few unit-test documenting and testing the schema validations. 
- [x] Update strategy-api docs

### CODE REVIEW

**Comment by sveisvei** [comment]
Got warning when starting up with this branch: `[WARN] Using unknown data type JSON`

**Comment by ivarconr** [comment]
unfortunately it is because `json` is not supported across different databases. It's started with the `client_applications` migration, but I didn't notice. 

Possible solutions:
1) use raw queries for data-type json columns 
2) adjust the db-migrate logger to only log errors (even the info is kind of anying)
3) don't do anything, it will only happen first time a new migration is executed.

File: lib/routes/strategy.js [hunk 1]
```diff
@@ -64,14 +63,10 @@ module.exports = function (app, config) {
...
-        req.checkBody('name', 'Name must match format ^[0-9a-zA-Z\\.\\-]+$').matches(/^[0-9a-zA-Z\\.\\-]+$/i);
-
-        const newStrategy = req.body;
-
-        validateRequest(req)
+        const data = req.body;
+        validateInput(data)
             .then(validateStrategyName)
-            .then(() => eventStore.store({
+            .then((newStrategy) => eventStore.store({
```
Comment (sveisvei) [inline lib/routes/strategy.js]: Call newStrategy => data?

**Comment by ivarconr** [comment]
Can be merged as soon as the `unleash-frontend` code is updated.

**Pull request merged**